File: apps/admin-web/src/app/shared/auth.models.ts
export interface User {
  id: string;
  email: string;
  displayName: string;
  roles: string[];
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  accessToken: string;
  expiresAtUtc: string;
  user: User;
}

export interface AuthState {
  accessToken: string;
  expiresAtUtc: string;
  user: User;
}

File: apps/api/HaberPlatform.Api/Models/AuthModels.cs
namespace HaberPlatform.Api.Models;

public record LoginRequest(string Email, string Password);

public record LoginResponse(
    string AccessToken,
    DateTime ExpiresAtUtc,
    UserDto User
);

public record UserDto(
    Guid Id,
    string Email,
    string DisplayName,
    string[] Roles
);

public record CreateUserRequest(
    string Email,
    string DisplayName,
    string TempPassword,
    string[] Roles
);

public record UserListDto(
    Guid Id,
    string Email,
    string DisplayName,
    bool IsActive,
    DateTime CreatedAtUtc,
    string[] Roles
);

